src/domain/flow.ts

```diff
@@ -1,22 +1,15 @@
 import { UploadResponse } from "@actions/artifact";
 import { CheckedOutNode } from "@bc/domain/checkout";
 import { ExecuteNodeResult } from "@bc/domain/execute-node-result";
-import { ExecutionPhase } from "@bc/domain/execution-phase";
 
 export type FlowResult = {
   checkoutInfo: CheckedOutNode[];
   artifactUploadResults: PromiseSettledResult<UploadResponse>[];
-  executionResult: {
-    [key in ExecutionPhase]: ExecuteNodeResult[]
-  }
+  executionResult: ExecuteNodeResult[][]
```

Comment:
sorry but I would create a new object here
```
executionResult: {
  before: ExecuteNodeResult[],
  commands: ExecuteNodeResult[],
  after: ExecuteNodeResult[]
}
```
wdyt?

Reply:
matrix could be tricky object type having this amazing typed TS language, wdyt?

Reply:
thats what I had before but now since we need to execute before, current and after per node I had to switch to the matrix one. If we separated out the before, current and after from each other then while printing execution errors we won't be able to do it per project. We will end up printing all before errors first then current errors and then after errors.